src/main/java/com/socialmediaraiser/twitterbot/impl/personalAnalyzer/ApiSearchHelper.java

```diff
@@ -2,118 +2,116 @@
+        if (!this.getUserId().equals(initialTweet.getAuthorId()) && !answeredByUserTweets.contains(initialTweet.getId())) {
+          System.out.print(".");
+          UserInteractions.UserInteraction userInteraction = userInteractions.get(tweet.getAuthorId());
+          userInteraction.incrementNbRepliesGiven();
+          answeredByUserTweets.add(initialTweet.getId());
         }
+      }
+    }
+    LOGGER.info("\n" + tweetWithReplies.size() + " replies given found, " + answeredByUserTweets.size() + " saved");
+  }
 
-        Map<String, List<String>> statusesAndAnswers = new HashMap<>();
-        int savedAnswers = 0;
+  /**
+   * Count the replies received by the user from others
+   * @param currentWeek if true, use the labs2 endpoint to only count the last 7 days, otherwise classical search within 30 days until D-7
+   * @return a map with tweet id as key and TweetInteraction as value
+   */
+  public Map<String, TweetInteraction> countRepliesReceived(boolean currentWeek) {
+    LOGGER.info("Counting replies received - currentWeek = " + currentWeek + " ...");
+    Stream<ITweet> tweetWithReplies = getReceivedReplies(currentWeek);
+    return tweetWithReplies.filter(tweet -> this.isUserInList(tweet.getAuthorId()))
+                           .map(tweet -> this.getTwitterClient().getInitialTweet(tweet, true))
+                           .filter(initialTweet -> this.getUserId().equals(initialTweet.getAuthorId()))
+                           .peek(initialTweet -> LOGGER.info("."))
+                           .groupBy(ITweet::getId)
+                           .map(this::foldTweets);
+  }
 
-        for(ITweet tweet : tweetWithReplies){
-            if(this.isUserInList(tweet.getAuthorId())){
-                ITweet initialTweet = this.getTwitterClient().getInitialTweet(tweet, true);
-                if(this.getUserId().equals(initialTweet.getAuthorId())){
-                    System.out.print(".");
-                    // if the initial tweet id is not on the map, we add it
-                    if(!statusesAndAnswers.containsKey(initialTweet.getId())){
-                        statusesAndAnswers.put(initialTweet.getId(), new ArrayList<>());
-                    }
-                    if(!statusesAndAnswers.get(initialTweet.getId()).contains(tweet.getAuthorId())) {
-                        UserInteractions.UserInteraction userInteraction = userInteractions.get(tweet.getAuthorId());
-                        userInteraction.incrementNbRepliesReceived();
-                        statusesAndAnswers.get(initialTweet.getId()).add(tweet.getAuthorId());
-                        savedAnswers++;
-                    }
-                }
-            }
+  private Tuple2<String, TweetInteraction> foldTweets(String id, Stream<ITweet> tweets) {
+    return Tuple.of(id,
+                    tweets.foldLeft(new TweetInteraction(),
+                                    (interaction, tweet) -> interaction.addAnswerer(tweet.getAuthorId())));
```

Comment:
@hypr2771 the answerer id we have to add is not the author of the initialTweet but the userId who answered to the tweet. I didn't manage to add it but I guess we would need another argument in the method 
```
  private Tuple2<String, TweetInteraction> foldTweets(String tweetId, String userId, Stream<ITweet> tweets) {
    return Tuple.of(tweetId,
                    tweets.foldLeft(new TweetInteraction(),
                                    (interaction, tweet) -> interaction.addAnswerer(userId)));
```

Reply:
Solved !! 
```
    return tweetWithReplies.filter(tweet -> this.isUserInList(tweet.getAuthorId()))
                           .filter(tweet -> this.getUserId().equals(
                               this.getTwitterClient().getInitialTweet(tweet, true).getAuthorId()))
                           .peek(initialTweet -> LOGGER.info("."))
                           .groupBy(tweet ->  this.getTwitterClient().getInitialTweet(tweet, true).getId())
                           .map((tweetId, tweets) -> this.foldTweets(tweetId, tweets));
```